lib/parallel_guest_migration_base.pm

```diff
@@ -1750,6 +1754,26 @@ sub check_peer_test_run {
     return $_peer_test_run_result;
 }
 
+=head2 filter_migration_tests
+
+Filters migration tests based on specified criteria.
+
+=cut
+
+sub filter_migration_tests {
+    my ($self, @migration_tests) = @_;
+    my @guest_migration_test;
```

Comment:
I think you'd better use named argument.
```
my ($self, %args) = @_;
```

All subroutines use named arguments in this module.